Products/ZenHub/PBDaemon.py

```diff
@@ -643,14 +643,30 @@ def connecting(self):
         """
         self.log.info("Attempting to connect to zenhub")
 
+    def getZenhubInstanceId(self):
+        """
+        Called after we connected to zenhub.
+        """
+
+        def callback(result):
+            self.log.info("Connected to the zenhub/%s instance", result)
+
+        def errback(result):
+            self.log.info("Unexpected error appeared while getting zenhub instance number %s", result)
+
+        d = self.perspective.callRemote('getHubInstanceId')
+        d.addCallback(callback)
+        d.addErrback(errback)
+        return d
+
     def gotPerspective(self, perspective):
         """
         This gets called every time we reconnect.
 
         @parameter perspective: Twisted perspective object
         @type perspective: Twisted perspective object
         """
-        self.log.info("Connected to ZenHub")
+        self.getZenhubInstanceId()
         self.perspective = perspective
```

Comment:
The `getZenhubInstanceId` call should be placed after setting the `perspective` attribute because `getZenhubInstanceId` depends on the `perspective` existing and having a valid object.